### PR TITLE
Z2m filter created devices

### LINKED ISSUE #1890: [zigbee2mqtt] Hide already added devices
Is it possible to hide devices that have already been added to Gladys by default? Maybe with an option to show them if needed. Because, especially on mobile, the list is so long that scrolling through it becomes quite a chore.

https://community.gladysassistant.com/t/ameliorations-dans-lui-de-lintegration-zigbee2mqtt/8364

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Fixes #1890 and #1655

### CODE REVIEW

File: server/services/zigbee2mqtt/lib/getDiscoveredDevices.js [hunk 1]
```diff
@@ -3,21 +3,27 @@ const { convertDevice } = require('../utils/convertDevice');
...
+  let devices = Object.values(this.discoveredDevices)
+    // Convert to Gladys device
+    .map((d) => convertDevice(d, this.serviceId))
+    .map((d) => {
+      const existingDevice = this.gladys.stateManager.get('deviceByExternalId', d.external_id);
+      // Merge with existing device.
+      return mergeDevices(d, existingDevice);
+    });
+
+  const { filterExisting } = filters;
```
Comment (Pierre-Gilles) [inline server/services/zigbee2mqtt/lib/getDiscoveredDevices.js]: Can you put this in snake_case (only in the REST API)

The API is snake_case only!

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a small feedback

File: server/services/zigbee2mqtt/lib/getDiscoveredDevices.js (hunk 1 above)
Comment (atrovato) [reply server/services/zigbee2mqtt/lib/getDiscoveredDevices.js]: Ok, done, see the last commit ;)
I'll have to change LANManager to do same change :)

**Comment by atrovato** [comment]
I made the requested change, I will do the same for LANManger later on another PR.

**Review by Pierre-Gilles — APPROVED** [review]
Good now!

**Pull request merged**